File: Capitulo02/selection_sort.py
#Function to find the index of the smallest element
def find_smallest(arr):
    smallest = arr[0]   #stores 1st item as the smallest value
    smallest_index = 0   #stores the index of the smallest value

    for i in range(1, len(arr)):
        if arr[i] < smallest:
            smallest = arr[i]
            smallest_index = i
    return smallest_index

#Main Selection Sort function
def selection_sort(arr):
    new_arr = []
    for i in range(len(arr)):   
        #Finds the smallest elemento in the array
        smallest = find_smallest(arr)  
        # ... pops it from the old array and appends to the new array
        new_arr.append(arr.pop(smallest))
    
    return new_arr

print(selection_sort([5, 3, 6, 2, 10]))

File: Capitulo02/SelectionSort.java
package Capitulo02;

import java.util.Arrays;

public class SelectionSort {
// Function to fin the index of the smallest element (starting from 'start)

    private static int findSmallest(int[] arr, int start) {
        int smallestIndex = start;

        for (int i = start +1; i < arr.length; i++) {
            if (arr[i] < arr[smallestIndex]) {
                smallestIndex = i;
            }
        }
        return smallestIndex;
    }
    
    // Mains Selection Sort function

    public static void selectionSort(int[] arr) {
        for (int i =0; i < arr.length; i++) {
            // Finds the smallest element in the remaining unsorted part.
        int smallestIndex = findSmallest(arr, i);
        
        // ... and swaps it with the current element
        int temp = arr[i];
        arr[i] = arr[smallestIndex];
        arr[smallestIndex] = temp;
        }
    }
    public static void main(String[] arts) {
        int[] myArray = {5, 3, 6, 2, 10};

        System.out.println("Original: " + Arrays.toString(myArray));

        selectionSort(myArray);

        System.out.println("Sorted: " + Arrays.toString(myArray));
    }

}
